### ISSUE TITLE
Support for Flicker Free Boot?

### ISSUE THREAD

**Issue by rinoandrejohnsen** [sddm/sddm#1518]
Hi! As I understand it the login manager (sddm in this case) must also support Flicker Free Boot for the entire experience to be good. GDM har this already implemented. Has any devs at here in the SDDM repo looked into it?

**Comment by dobey** [comment]
> GDM har this already implemented.

I'm not sure it especially has anything to do with GDM itself. So what does GDM have exactly?

**Comment by aleixpol** [comment]
It's not something to implement itself, as far as I know it's about properly orchestrating the different processes to start in ways that the user does not perceive it.

I'd recommend to look into this issue after we've been able to release what we have now and we can start dropping some features as currently it's a big juggle.

**Comment by dobey** [comment]
> It's not something to implement itself, as far as I know it's about properly orchestrating the different processes to start in ways that the user does not perceive it.

Yes, I know, that's why I'm curious what gdm has to do with it if anything.

For Ubuntu Touch, what we have is a system compositor which runs at early boot, and then the login manager and session both run as clients of the system compositor. This is used instead of plymouth, and the phones are booted without kernel console, and no virtual terminals. This gives the smoothest possible boot on those devices. The same thing can be done on PC or phones running Plasma Mobile, or other devices running Plasma, as well.

**Comment by rinoandrejohnsen** [comment]
From my understanding, this is all about the framebuffer handover from Plymouth. GDM sets up it's own framebuffer and scans the framebuffer from Plymouth and continues to show this until Login/Desktop has been reached.

This makes sure that the boot logo from EFI to Login/Desktop is shown as a continued screen across several steps.

If SDDM can keep this framebuffer, then the Ksplash can take it further and give a continuous experience for those with auto-login form EFI to Desktop.

**Comment by zentasumu** [comment]
OS: Arch Linux x86_64 
Kernel: 6.6.10-arch1-1
CPU: 12th Gen Intel i5-12500H (16) @ 4.500GHz 
GPU: Intel Alder Lake-P GT2 [Iris Xe Graphics] 
GPU: NVIDIA GeForce RTX 3060 Mobile / Max-Q

For wayland, I have tested SDDM development branch with weston 13.0 and the framebuffer handover from Plymouth appeared smooth, but using kwin_wayland or sway as the compositor shows a black screen in-between.

Maybe this is indeed more on the compositor side than the display manager.

I did also try to hackishly match how GDM handles VT switching as I saw some functions are called in a different order, and proceeded to use kwin_wayland with it, but the issue remains.
Though I am unsure if I was looking at the right place in the code to experiment in as I only just began diving in this.
